portfolio/src/main/webapp/script.js

```diff
@@ -119,3 +119,37 @@ function hideOnMouseout(hoveredItem) {
     tag.style.filter = 'brightness(100%)';
   }
 }
+
+/**
+ * Fetches the message from the server /data and adds it to the DOM.
+ */
+function getMessage() {
+  const responsePromise = fetch('/data');
+  
+  // Passes responsePromise into handleResponse() after 
```

Comment:
No need for this comment as the code explains itself. Use comments where the logic might be extremely complex. A lot of the times good method naming and separation of concerns helps in describing the code clearly.

Reply:
Done!